adapters/ix/ix.go

```diff
@@ -100,6 +127,129 @@ func (a *IxAdapter) MakeRequests(request *openrtb2.BidRequest, reqInfo *adapters
 	return append(requests, multiSizeRequests...), errs
 }
 
+func handleMultiImpInSingleRequest(request *openrtb2.BidRequest, a *IxAdapter) ([]*adapters.RequestData, []error) {
+	nImp := len(request.Imp)
+	if nImp > a.maxRequests {
+		request.Imp = request.Imp[:a.maxRequests]
+		nImp = a.maxRequests
+	}
+	// Multi-size banner imps are split into single-size requests.
+	// The first size imp requests are added to the first slice.
+	// Additional size requests are added to the second slice and are merged with the first at the end.
+	requests := make([]*adapters.RequestData, 0, a.maxRequests)
+	multiSizeRequests := make([]*adapters.RequestData, 0, a.maxRequests-1)
+	errs := make([]error, 0)
+
+	if err := BuildIxDiag(request); err != nil {
+		errs = append(errs, err)
+	}
+
+	headers := http.Header{
+		"Content-Type": {"application/json;charset=utf-8"},
+		"Accept":       {"application/json"}}
+
+	siteIds := make(map[string]bool)
+	originalImps := make([]openrtb2.Imp, 0, len(request.Imp))
+	additionalMultiSizeImps := make([]openrtb2.Imp, 0)
+	requestCopy := *request
+	for _, imp := range requestCopy.Imp {
+		if err := parseSiteId(&imp, siteIds); err != nil {
+			errs = append(errs, err)
+			continue
+		}
+
+		if err := moveSid(&imp); err != nil {
+			errs = append(errs, err)
+		}
+
+		if imp.Banner != nil {
+			bannerCopy := *imp.Banner
+
+			if len(bannerCopy.Format) == 0 && bannerCopy.W != nil && bannerCopy.H != nil {
+				bannerCopy.Format = []openrtb2.Format{{W: *bannerCopy.W, H: *bannerCopy.H}}
+			}
+
+			if len(bannerCopy.Format) == 1 {
+				bannerCopy.W = openrtb2.Int64Ptr(bannerCopy.Format[0].W)
+				bannerCopy.H = openrtb2.Int64Ptr(bannerCopy.Format[0].H)
+			}
+
+			if len(bannerCopy.Format) > 1 {
+				formats := bannerCopy.Format
+				// Creating additional imp from multiple formats. First format is part of original imp. Rest format will be converted to individual single imp & later request.
+				for iFmt := range formats {
+					additionalBannerCopy := *imp.Banner
+					additionalBannerCopy.Format = formats[iFmt : iFmt+1]
+					additionalBannerCopy.W = openrtb2.Int64Ptr(additionalBannerCopy.Format[0].W)
+					additionalBannerCopy.H = openrtb2.Int64Ptr(additionalBannerCopy.Format[0].H)
+					if iFmt == 0 {
+						bannerCopy = additionalBannerCopy
+					} else {
+						imp.Banner = &additionalBannerCopy
+						additionalMultiSizeImps = append(additionalMultiSizeImps, imp)
+					}
+				}
+			}
+
+			imp.Banner = &bannerCopy
+		}
+		originalImps = append(originalImps, imp)
+	}
+
+	if requestCopy.Site != nil {
+		site := *requestCopy.Site
+		if site.Publisher == nil {
+			site.Publisher = &openrtb2.Publisher{}
+		}
+		if len(siteIds) == 1 {
+			for siteId := range siteIds {
+				site.Publisher.ID = siteId
+			}
```

Comment:
This is looping through and setting the publisher id each time, such that the last siteId in the array always wins. You can avoid the loop entirely and access the last item in the slice directly.